llvm/lib/Transforms/Scalar/CorrelatedValuePropagation.cpp

```diff
@@ -1227,6 +1257,7 @@ CorrelatedValuePropagationPass::run(Function &F, FunctionAnalysisManager &AM) {
   if (!Changed) {
     PA = PreservedAnalyses::all();
   } else {
+    assert(DT->verify());
```

Comment:
We shouldn't unconditionally verify.

Reply:
Isn't it guarded by the `NDEBUG` marco?

Should I use the fast-path if `LLVM_ENABLE_EXPENSIVE_CHECKS` is OFF?
```
#if defined(EXPENSIVE_CHECKS)
  assert(DT.verify(DominatorTree::VerificationLevel::Full));
#else
  assert(DT.verify(DominatorTree::VerificationLevel::Fast));
#endif // EXPENSIVE_CHECKS
```